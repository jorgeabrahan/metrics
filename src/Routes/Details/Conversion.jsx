/* eslint-disable import/no-extraneous-dependencies */
import PropTypes from 'prop-types';

const Conversion = ({ name, value, isDarker }) => (
  <div className={`conversion ${isDarker ? '' : 'conversion--ruber'}`}>
    <h3 className="conversion__name">{name}</h3>
    <p className="conversion__value">{value}</p>
  </div>
);

Conversion.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.number.isRequired,
  isDarker: PropTypes.bool.isRequired,
};

export default Conversion;
