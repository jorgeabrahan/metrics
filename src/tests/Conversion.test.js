import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Conversion from '../Routes/Details/Conversion';
import wrap from './testResources';

describe('Tests for <Conversion /> component', () => {
  const conversion = (
    <Conversion
      name="usd"
      value={120}
      isDarker={false}
    />
  );

  it('Should render the content correctly', () => {
    render(wrap(conversion));
    expect(screen.getByText('usd')).not.toBeNull();
    expect(screen.getByText('120')).not.toBeNull();
  });

  it('Should match the snapshot', () => {
    expect(renderer.create(wrap(conversion)).toJSON()).toMatchSnapshot();
  });
});
