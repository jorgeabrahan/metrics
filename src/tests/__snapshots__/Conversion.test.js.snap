// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Conversion /> component Should match the snapshot 1`] = `
<div
  className="conversion "
>
  <h3
    className="conversion__name"
  >
    usd
  </h3>
  <p
    className="conversion__value"
  >
    120
  </p>
</div>
`;
